app/src/androidTest/java/org/mozilla/tv/firefox/ui/BasicNavigationTest.kt

```diff
@@ -128,13 +128,32 @@ class BasicNavigationTest {
             assertNotEquals(preReloadTitle, getPageTitle())
         }
     }
+
     @Test
     fun exitAppByBackButtonTest() {
         navigationOverlay {
             remoteBack()
             assertActivityFinishing(activityTestRule)
         }
     }
+
+    @Test
+    fun navigateThenExitAppByBackButtonTest() {
+        val server = MockWebServer().apply {
```

Comment:
Since we might be using this for multiple tests in this file, I wonder if there are any savings/clarity by setting this up in one place, but a) not every test uses it, and b) since we clear app state in between test runs, I guess it doesn't matter because the server object can't persist?

Reply:
I think we'd ideally just have a really nice helper that sets things up, but we've tried that and, to be perfectly honest, I think we've landed on a state where it's just confusing.  The requirements are a little bit different between each test.  I haven't really figured out how I want this to work, but I share your concerns.